src/tool_operate.c

```diff
@@ -697,45 +697,59 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
   return result;
 }
 
+/* helper function to ensure input ends in val_to_ensure */
+static void ensure_trailing(char **input, const char val_to_ensure)
+{
+  if(*input && **input) {
+    size_t len = strlen(*input);
+    if(((*input)[len - 1] != val_to_ensure)) {
+      char *composed_input = aprintf("%s%c", *input, val_to_ensure);
+      if(composed_input) {
+        Curl_safefree(*input);
+        *input = aprintf("%s", composed_input);
+        Curl_safefree(composed_input);
+      }
```

Comment:
```suggestion
      char *composed_input = aprintf("%s%c", *input, val_to_ensure);
      Curl_safefree(*input);
      *input = composed_input;
```

(should be equivalent?)

Or even:

```c
      *input = Curl_saferealloc(*input, len+2); // needs some casting
      if(*input) {
         input[len] = val_to_ensure;
         input[len+1] = 0;
      }
```

But, I'm not sure if that violates some rule of curl's coding style.

Reply:
I think that's fine, i see more of those constructions in curl.
Thank you for this hint!  This idea works just fine and saves a temp copy!